test-alerts.cpp

```diff
@@ -1,8 +1,57 @@
 #define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
 
 #include "test/catch.hpp"
-#include "typewise-alert.h"
+#include "Alerter.hpp"
+#include <iostream>
 
 TEST_CASE("infers the breach according to limits") {
-  REQUIRE(inferBreach(12, 20, 30) == TOO_LOW);
+  Breach testBreach;
+  REQUIRE(testBreach.inferBreach(10, 20, 30) == BreachType::TOO_LOW);
+  REQUIRE(testBreach.inferBreach(15, 20, 30) == BreachType::TOO_LOW);
+  REQUIRE(testBreach.inferBreach(20, 20, 30) == BreachType::NORMAL);
+  REQUIRE(testBreach.inferBreach(25, 20, 30) == BreachType::NORMAL);
+  REQUIRE(testBreach.inferBreach(30, 20, 30) == BreachType::NORMAL);
+  REQUIRE(testBreach.inferBreach(35, 20, 30) == BreachType::TOO_HIGH);
 }
+
+TEST_CASE("identify breach type for Passive cooling and send alert"){
+  EmailAlerter testEmailAlerter(CoolingType::PASSIVE_COOLING, "Siemens");
```

Comment:
Code calling classifyTemperatureBreach() should be generic. What this means is, the method invocation should be on Alerter object and not EmailAlerter or controllerAlerter, since the primary reason for introducing inheritance is to abstract type of alerter object.